### ISSUE TITLE
vc++11 folly build error

### ISSUE THREAD

**Issue by Mark-Joy** [jbandela/cppcomponents#4]
Hello jbandela,

Sorry to bring this question here as I couldn't open issue on your VC++11 folly project: https://github.com/jbandela/folly/tree/vc11

I try to build with Visual Studio 2015 but I got the error  'folly-config.h': No such file or directory
Steps I did:
1> Use generate_escape_tables.py to generate EscapeTables.cpp
2> Use generate_format_tables.py to generate FormatTables.cpp
3> Use generate_varint_tables.py to generate GroupVarintTables.cpp
4> Change "#include x86intrin.h" to "#include intrin.h" in GroupVarintTables.cpp
5> Include folder "folly-vc11" to "Include Directories" in VS2015
6> Include folder "folly-vc11\double-conversion\src" to "Include Directories" in VS2015
7> Include folder "boost_1_62_0" to "Include Directories" in VS2015
8> Build

I got the below errors when building. How to solve it?

Thank you in advance

`1>------ Build started: Project: follyvc11lib, Configuration: Release Win32 ------
1>  FormatTables.cpp
1>F:\Programming_Language\C++\1_Libs\folly-vc11\folly/Malloc.h(54): fatal error C1083: Cannot open include file: 'folly-config.h': No such file or directory
1>  GroupVarintTables.cpp
1>..\..\..\folly\build\GroupVarintTables.cpp(264): warning C4838: conversion from 'unsigned __int64' to 'char' requires a narrowing conversion
1>..\..\..\folly\build\GroupVarintTables.cpp(264): warning C4305: 'initializing': truncation from 'unsigned __int64' to 'char'
1>..\..\..\folly\build\GroupVarintTables.cpp(264): warning C4309: 'initializing': truncation of constant value
1>  Conv.cpp
1>F:\Programming_Language\C++\1_Libs\folly-vc11\folly/Malloc.h(54): fatal error C1083: Cannot open include file: 'folly-config.h': No such file or directory
1>  dynamic.cpp
1>F:\Programming_Language\C++\1_Libs\folly-vc11\folly/Malloc.h(54): fatal error C1083: Cannot open include file: 'folly-config.h': No such file or directory
1>  Format.cpp
1>F:\Programming_Language\C++\1_Libs\folly-vc11\folly/Malloc.h(54): fatal error C1083: Cannot open include file: 'folly-config.h': No such file or directory
1>  json.cpp
1>F:\Programming_Language\C++\1_Libs\folly-vc11\folly/Malloc.h(54): fatal error C1083: Cannot open include file: 'folly-config.h': No such file or directory
1>  Malloc.cpp
1>f:\programming_language\c++\1_libs\folly-vc11\folly\Malloc.h(54): fatal error C1083: Cannot open include file: 'folly-config.h': No such file or directory
1>  Range.cpp
1>F:\Programming_Language\C++\1_Libs\folly-vc11\folly/Malloc.h(54): fatal error C1083: Cannot open include file: 'folly-config.h': No such file or directory
1>  String.cpp
1>F:\Programming_Language\C++\1_Libs\folly-vc11\folly/Malloc.h(54): fatal error C1083: Cannot open include file: 'folly-config.h': No such file or directory
1>  Unicode.cpp
1>F:\Programming_Language\C++\1_Libs\folly-vc11\folly/Malloc.h(54): fatal error C1083: Cannot open include file: 'folly-config.h': No such file or directory
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========
`

**Comment by jbandela** [comment]
Hi Mark-Joy,

To be honest the fork was created for Visual C++ 2012 which did not even
have variadic template or unrestricted unions. Visual C++ 2015 pretty much
supports C++11. I actually think it would be easier and more robust to
adapt current folly to Visual C++ 2015. The biggest thing you would have to
do is remove a few gcc extensions mainly where they are using jemalloc

John

**Comment by Mark-Joy** [comment]
Thank you for your project :)
I just generated the file folly-config.h and got your project built successfully under VC++ 2015. I need only the folly/format lib (latest version is the best but old version is still fine to me).
Tried to combine latest folly under MSYS2 and MINGW-W64, but I got some errors.
For now I will stick to your project only. Will try to port latest folly to VC++ 2015 if I have some free time.
